src/nvector/mpiplusx/nvector_mpiplusx.c

```diff
@@ -14,23 +14,26 @@
  * This is the implementation file for the MPIPlusX NVECTOR.
  * -----------------------------------------------------------------*/
 
-#include <nvector/nvector_mpimanyvector.h>
 #include <nvector/nvector_mpiplusx.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <sundials/sundials_math.h>
+#include <sundials/priv/sundials_context_impl.h>
+#include <sundials/priv/sundials_errors_impl.h>
+#include <sundials/sundials_core.h>
+
+#include "sundials/priv/sundials_errors_impl.h"
+#include "sundials/sundials_errors.h"
 
 #define MPIPLUSX_LOCAL_VECTOR(v) (N_VGetSubvector_MPIManyVector(v, 0))
 
 N_Vector N_VMake_MPIPlusX(MPI_Comm comm, N_Vector X, SUNContext sunctx)
 {
+  SUNFunctionBegin(sunctx);
   N_Vector v;
 
-  if (X == NULL) { return NULL; }
+  SUNAssertNull(X, SUN_ERR_ARG_CORRUPT);
 
   v = NULL;
-  v = N_VMake_MPIManyVector(comm, 1, &X, sunctx);
-  if (v == NULL) { return NULL; }
+  v = N_VMake_MPIManyVector(comm, 1, &X, SUNCTX_);
```

Comment:
We want to refer to `SUNCTX_` for all uses of the SUNContext object in our functions except in the `SUNFunctionBegin` macro (this is encoded in the rules).  The reason for this is to ensure that the SUNContext accessed is the same one throughout a function. E.g., to avoid something like this (this is contrived but there are places where it comes up):

```c
SUNErrCode N_VLinearCombination_Serial(int nvec, sunrealtype* c,
                                                N_Vector* V, N_Vector z)
{
   N_VMake_Serial(length, V[0]->sunctx);
   N_VMake_Serial(length, z->sunctx);
}
```

The macro also allows us to avoid dereferencing an object over and over to access the SUNContext, although this is likely negligible. Now, in the case of `N_VMake_MPIManyVector`, the `SUNContext` is an argument to function directly so its not really a problem. However, making exceptions to the rules only makes them more complicated.